jerry-core/ecma/builtin-objects/ecma-builtin-helpers.c

```diff
@@ -219,44 +219,25 @@ ecma_builtin_helper_get_to_locale_string_at_index (ecma_object_t *obj_p, /**< th
   }
 
   ecma_string_t *ret_string_p = NULL;
-  ecma_object_t *index_obj_p = ecma_get_object_from_value (index_obj_value);
-  ecma_value_t to_locale_value = ecma_op_object_get_by_magic_id (index_obj_p, LIT_MAGIC_STRING_TO_LOCALE_STRING_UL);
 
-  if (ECMA_IS_VALUE_ERROR (to_locale_value))
-  {
-    goto cleanup;
-  }
-
-  if (!ecma_op_is_callable (to_locale_value))
-  {
-    ecma_free_value (to_locale_value);
-    ecma_raise_type_error (ECMA_ERR_MSG ("'toLocaleString' is missing or not a function."));
-    goto cleanup;
-  }
-
-  ecma_object_t *locale_func_obj_p = ecma_get_object_from_value (to_locale_value);
 #if ENABLED (JERRY_ESNEXT)
-  ecma_value_t index_parameter = index_value;
+  ecma_value_t call_value = ecma_op_invoke_by_magic_id (index_value, LIT_MAGIC_STRING_TO_LOCALE_STRING_UL, NULL, 0);
 #else /* !ENABLED (JERRY_ESNEXT) */
-  ecma_value_t index_parameter = index_obj_value;
+  ecma_value_t call_value = ecma_op_invoke_by_magic_id (index_obj_value, LIT_MAGIC_STRING_TO_LOCALE_STRING_UL, NULL, 0);
```

Comment:
This seems weird. The `index_obj_value` is already a ToObject result  and the invoke will also call the ToObject method. Is this really correct?